asab/storage/mongodb.py

```diff
@@ -60,6 +133,15 @@ def __init__(self, app, service_name, config_section_name='asab:storage'):
 
 		assert self.Database is not None
 
+		# For the transactions, session is needed
+		self._current_session = None
```

Comment:
You cannot do it this way - if there is a concurrently / asynchronously running task, this will be mixed.

I'm afraid that the whole design of this MR must be done differently.